aws/internal/service/workspaces/waiter/status.go

```diff
@@ -34,7 +34,7 @@ func WorkspaceState(conn *workspaces.WorkSpaces, workspaceID string) resource.St
 		}
 
 		if len(output.Workspaces) == 0 {
-			return nil, "", nil
+			return output, workspaces.WorkspaceStateTerminated, nil
```

Comment:
We consider an empty list in output as a real workspace termination sign. The `workspaces.WorkspaceStateTerminated` state is checked right after the `DescriobeWorkspaces` call in `Read` func.